### ISSUE TITLE
Add functionality to append supplied String (or Supplier<String>) to default error message

### ISSUE THREAD

**Issue by nick-u-riis** [assertj/assertj-core#2011]
#### Summary
Currently we can call withFailMessage with either a String or Supplier<String>. This is nice but the default error message can be useful **in addition** to a custom error message. Please add this kind of method with both signatures.

Thanks for the consideration!

#### Example

```java
// logs "Inconceivable!" only
Assertions.assertThat("string").withFailMessage("Inconceivable!").contains("sub");
/** Would log the following:
Expecting:
 <"string">
to contain:
 <"sub">
Inconceivable!
 */
Assertions.assertThat("string").withAdditionalFailMessage("no").contains("sub");
```

**Comment by nach-o-man** [comment]
@nick-u-riis this can already be done with 

`Assertions.assertThat("string").as(() -> "Inconceivable!").contains("sub");`

Output example:

> java.lang.AssertionError: [Inconceivable!] 
Expecting actual:
  "string"
to contain:
  "sub" 

>

**Comment by joel-costigliola** [comment]
Thanks @nach-o-man, indeed `as` can be used to give some context explanation to the following assertion.

@nick-u-riis does that address your concen?

**Comment by WhiteLittleDuck** [comment]
Hi! I think it would be better if the solution given by @nach-o-man can be wrapped in a method in AbstractAssert. So user can use the method such as `Assertion.assertThat("string").withAdditionalFailMessage("Inconceivable!").contains("sub")` which seems more interpretable :)

**Comment by nach-o-man** [comment]
@WhiteLittleDuck as far as [javadoc says](https://github.com/assertj/assertj-core/blob/cc65ca7a04586de803a1df7e386a98a438b607b1/src/main/java/org/assertj/core/api/AbstractAssert.java#L646) about withFailMessage method 

>    @param newErrorMessage the error message that will replace the default one provided by Assertj.

so there is no _addition_ to anything there is only replacement. So I believe using **as** is pretty much solution to the problem.

**Comment by scordio** [comment]
The existing `as()` and `withFailMessage()` should give already some degree of flexibility. Probably appending text to existing messages would not be enough for general usage. We might need something smarter than that, but I'd wait for a concrete use case before thinking about an additional API to manipulate error messages.

**Comment by vlsi** [comment]
I agree `withFailMessage` might indeed confuse people a lot, and I think rare when `withFailMessage` is the right choice over `as`.
For instance, see https://github.com/openrewrite/rewrite-testing-frameworks/issues/234 where they currently use `withFailMessage` instead of `as`.

I wonder if `withFailMessage` could be deprecated or renamed so that people do not immediately use it, especially when the only thing they need is to "append message"
